test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2434,6 +2435,22 @@ public static <T extends Throwable> T expectThrows(Class<T> expectedType, Reques
         );
     }
 
+    /**
+     * Same as {@link #runInParallel(int, IntConsumer)} but also attempts to start all tasks at the same time by blocking execution on a
+     * barrier until all threads are started and ready to execute their task.
+     */
+    public static void startInParallel(int numberOfTasks, IntConsumer taskFactory) throws InterruptedException {
+        final CyclicBarrier barrier = new CyclicBarrier(numberOfTasks);
+        runInParallel(numberOfTasks, i -> {
+            try {
+                barrier.await();
```

Comment:
I think I'd use a `safeAwait` here to save the try/catch noise and (preserve the interrupt flag properly fwiw).

Reply:
++ done